src/base/digest32.h

```diff
@@ -75,22 +82,27 @@ class Digest32
             return {};
 
         Digest32 result;
-        result.m_valid = true;
-        result.m_hashString = digestString;
-        result.m_nativeDigest.assign(raw.constData());
+        result.m_dataPtr->valid = true;
+        result.m_dataPtr->hashString = digestString;
+        result.m_dataPtr->nativeDigest.assign(raw.constData());
 
         return result;
     }
 
     QString toString() const
     {
-        return m_hashString;
+        return m_dataPtr->hashString;
     }
 
 private:
-    bool m_valid = false;
-    UnderlyingType m_nativeDigest;
-    QString m_hashString;
+    struct Data : public QSharedData
```

Comment:
`struct Data final : public QSharedData`?

Reply:
> `struct Data final : public QSharedData`?

Why?
QSharedData even has no virtual methods.
And Data is declared privately so no one can inherit it.